File: backend/mcp-code-intelligence-python/src/mcp_code_intel/memory/schema_v4.py
"""Schema V4 — Agent identification for knowledge entries.

Adds agent_name column to track which agent created each entry.
"""

# Add agent_name column to knowledge_entries
SCHEMA_V4_AGENT_NAME_ALTER = [
    "ALTER TABLE knowledge_entries ADD COLUMN agent_name TEXT DEFAULT NULL",
]

SCHEMA_V4_AGENT_NAME_INDEX = """
CREATE INDEX IF NOT EXISTS idx_ke_agent_name ON knowledge_entries(agent_name)
"""

File: backend/mcp-code-intelligence-kotlin/src/main/kotlin/com/codeintel/memory/schema/SchemaV4.kt
/** Schema V4 — Agent identification for knowledge entries. */
package com.codeintel.memory.schema

/**
 * V4 migration: adds agent_name column to track which agent created each entry.
 */
object SchemaV4 {

    /** Add agent_name column to knowledge_entries. */
    val AGENT_NAME_ALTER = listOf(
        "ALTER TABLE knowledge_entries ADD COLUMN agent_name TEXT DEFAULT NULL"
    )

    val AGENT_NAME_INDEX = """
        CREATE INDEX IF NOT EXISTS idx_ke_agent_name ON knowledge_entries(agent_name)
    """.trimIndent()
}
